### DISCUSSION TITLE
I wanna know how to transfer the Task to internal.Task?

### DISCUSSION THREAD

**Discussion by RyanChengStar** [MarioCarrion/todo-api-microservice-example#13] [Q&A] [answered]
Hi,Mario.I'm the beginner. I add a function to list all tasks, but i can't return [ ]internal.Task just return [ ]Task, so what can I do for transfering them? thx.

**Comment by MarioCarrion** [comment, marked as answer]
Hello @RyanChengStar 

I assume you mean something [like this](https://github.com/MarioCarrion/todo-api-microservice-example/blob/b5783586c14d426a555c0ca14ad8873d51f16c1d/internal/rest/task.go#L177)? but instead returning **all** the records.

I'd do something similar to that method `TaskHandler.search` where the internal type is converted into the one in the `rest` package, so this means in the `rest` package the interface type `TaskService` would define a new method called `All` (or similar) that is implemented by `postgres.Task` which returns _all_ the values from the datastore, this method would called by a new method `TaskHandler.all`  in the `rest` package.

I hope that helps

**Reply by RyanChengStar** [reply]
@MarioCarrion thx for specific answer!  I've seen “TaskHandler.search”  today, maybe I can  transfer them now. Thanks for sharing this good project! Your posts are beneficial for me, rushing you into updating them！
